compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -130,7 +114,7 @@ pub(crate) fn optimize_into_acir(
     let ssa_gen_span = span!(Level::TRACE, "ssa_generation");
     let ssa_gen_span_guard = ssa_gen_span.enter();
 
-    let ssa = SsaBuilder {
+    let mut ssa = SsaBuilder {
```

Comment:
Strange that using `&mut` somehow makes the `.par_bridge` used in the checks work, even though it shouldn't actually need mutation.

Reply:
Yeah it's some interior mutability issue which having a mutable reference allows us to sidestep.